docs/root/configuration/access_log.rst

```diff
@@ -29,18 +29,47 @@ are noted.
 
 The following command operators are supported:
 
+.. _config_access_log_format_start_time:
+
 %START_TIME%
   HTTP
     Request start time including milliseconds.
 
   TCP
     Downstream connection start time including milliseconds.
 
-  START_TIME can be customized using a `format string <http://en.cppreference.com/w/cpp/io/manip/put_time>`_, for example:
+  START_TIME can be customized using a `format string <http://en.cppreference.com/w/cpp/io/manip/put_time>`_.
+  In addition to that, START_TIME also accepts following specifiers:
 
-.. code-block:: none
+  +------------------------+-------------------------------------------------------------+
+  | Specifier              | Explanation                                                 |
+  +========================+=============================================================+
+  | ``%s``                 | The number of seconds since the Epoch                       |
+  +------------------------+-------------------------------------------------------------+
+  | ``%f``, ``%[1-9]f``    | Fractional seconds digits, default is 9 digits (nanosecond) |
+  |                        +-------------------------------------------------------------+
+  |                        | - ``%3f`` millisecond (3 digits)                            |
+  |                        | - ``%6f`` microsecond (6 digits)                            |
+  |                        | - ``%9f`` nanosecond (9 digits)                             |
+  +------------------------+-------------------------------------------------------------+
+
+  Examples of formatting START_TIME is as follows:
+
+  .. code-block:: none
+
+    %START_TIME(%Y/%m/%dT%H:%M:%S%z %s)%
+
+    # To include millisecond fraction of the second (.000 ... .999). E.g. 1527590590.528.
+    %START_TIME(%s.%3f)%
+
+    %START_TIME(%s.%6f)%
+
+    %START_TIME(%s.%9f)%
+
+  .. attention::
 
-  %START_TIME(%Y/%m/%dT%H:%M:%S%z %s)%
+    Do not include **?** in START_TIME's specifier, since it is used internally as placeholder
```

Comment:
e.g. given `%Y-%m-%dT%H:%M:%s.%3f` and `%Y-%m-%dT%H:%M:%s.%1f:%S:%6f`. 

For the first one, `%Y-%m-%dT%H:%M:%s.%3f`, we need to know the position of the subsecond after `%Y-%m-%dT%H:%M:%s.` is rendered. 

While for the second one, after we knew the position of the first subsecond (`%1f`) we need to know the position of the second subsecond (`%6f`) after `%Y-%m-%dT%H:%M:%s.%1f:%S` is rendered. This condition couldn't be tackled by simply recording the position without first rendering the previous part of the string before the subsecond specifier and track the position for correcting the position of the first subsecond specifier for the second one.

Reply:
You didn't need to do this before I think, because you were able to replace with a sequence like `???` where the length matched the N in `%Nf`. I think you can do this still here. You can do a regex find to locate the left-most `%Nf`. Record the position and width. Then, do a regex replace of the left-most `%Nf`, replacing with `???` of the appropriate length.

The advantage of doing it this way is you do all the heavy lifting with find operations at access log formatter construction time, rather than on the data path.

Reply:
@htuch yes, sorry. I think I misunderstood your previous comment. And I'm afraid to spend more than 5 lines of code budget 😄